### ISSUE TITLE
Unable to connect to server

### ISSUE THREAD

**Issue by JoshKPatterson** [mircokroon/minecraft-world-downloader#20]
So I run the command and the program clearly executes fine. When I'm instructed to connect to localhost, I try but I get an error in the console that goes
`Cannot connect to [SERVER IP]. The server may be down or on a different address.`
I know neither of the latter are the case, as I was able to personally connect to the server without issue.

Am I doing this wrong? I'm trying to connect to a minecraft server with IP 127.0.0.1:25565 when I'm prompted to connect to localhost

**Comment by ChillerDragon** [comment]
Could you maybe give more details? Like your operating system minecraft version and the address of the server you try to reach?

**Comment by leandrotlz** [comment]
If you are running a server in localhost and you also want to run the world-downloader on localhost, you need to assign it a different port. Add `--local-port 25566` and then connect Minecraft to localhost:25566 (or another unused port).

**Comment by JoshKPatterson** [comment]
After some more testing I've come to the conclusion that it was in fact the specific server I was trying to connect to. I'm able to use the program on some other, more popular servers without any issue. The only mods I know for sure the server to have is CraftBukkit.

**Comment by ChillerDragon** [comment]
@JoshKPatterson yea feel free to mail me the ip at chillerdragon@gmail.com but I probably won't be able to debug it. So maybe you could try to figure out some differences between the broken server and working servers and then mention them here.

**Comment by JoshKPatterson** [comment]
@ChillerDragon I've sent you an email containing the server IP. Thank you

**Comment by Jkmcameron** [comment]
Same issue, any progress?

**Comment by mircokroon** [comment]
I assume the original issue was also related to not specifying the port correctly. This could maybe be supported in the future but I don't think it's a high priority.

**Comment by JoshKPatterson** [comment]
@mircokroon it was not related to specifying the port. as I mentioned in previous comments it was a problem with the specific server, I would assume some kind of mod is causing the issue. I tried to specify the port with --port but that didn't cause any change to the issue.

**Comment by ChillerDragon** [comment]
Yes the whole port thing was confusing. It is a regular non local server on standard port. I also tested the downloader on my system and it did not work either. I assumed it was some sort of dns issue but also connecting to the ip did not work. I tried contacted the admin a while ago but seems like I did not reach him.

But since the ip is not public im not sure how much sense it makes to keep a  unreproducible issue open.
